pandas/core/series.py

```diff
@@ -4056,7 +4056,9 @@ def apply(
             Python function or NumPy ufunc to apply.
         convert_dtype : bool, default True
             Try to find better dtype for elementwise function results. If
-            False, leave as dtype=object.
+            False, leave as dtype=object. Note that conversion does not
+            happen for extension array dtypes which have a map method
```

Comment:
```suggestion
            happen for extension array dtypes, such as Categorical.
```

Reply:
There are different behaviors for different extension array dtypes - using convert_dtype=False with `Series([1], dtype=pd.Int64Dtype())` and `Series([1], dtype="category")` results in object and category dtypes respectively. Agreed on not mentioning the map method here, but it seems better to me to spell out which pandas dtypes will not come out as object when convert_dtype is False.